cmd/quickstart.go

```diff
@@ -102,12 +106,53 @@ func quickstartExec(ctx context.Context, flags QuickstartFlags) error {
 		return errors.Wrapf(err, "failed to validate workflow file")
 	}
 
-	outDir := flags.OutDir
+	outDir := workingDir
+	if flags.OutDir != "" {
+		outDir = flags.OutDir
+	}
+	var targetType string
 	for _, target := range quickstartObj.WorkflowFile.Targets {
-		if outDir == "" {
-			outDir = workingDir + "/" + defaultOutDir(target.Target)
-			break
+		targetType = target.Target
+		break
+	}
+
+	var isUncleanDir bool
+	if entry, err := os.ReadDir(workingDir); err == nil {
+		for _, e := range entry {
+			if !strings.HasPrefix(e.Name(), ".") {
+				isUncleanDir = true
+				break
+			}
+		}
+	}
+
+	if (isUncleanDir && outDir == workingDir) || (targetType == "terraform" && !strings.HasPrefix(filepath.Base(outDir), "terraform-provider")) {
+		promptedDir := "."
+		if outDir != workingDir {
+			promptedDir = outDir
+		}
+		description := "The option provided defaults to the current root directory."
```

Comment:
I don't quite understand this. Seems weird to say when i haven't provided an option yet, or am in the process of changing it?

Reply:
The default option we have provided maps to the current root directory.